Description: fixes issues with illegal windowsize hints
Author: Gaspar Chilingarov
Origin: upstream
Applied-Upstream: https://repo.or.cz/wmaker-crm.git/commitdiff/6668715402a5d8e2ecda6702076a06bf8988721e
Last-Update: 2018-11-25
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git src/window.c src/window.c
index cb4ff4e..195c3e6 100644 (file)
--- src/window.c
+++ src/window.c
@@ -2087,6 +2087,14 @@ void wWindowConfigure(WWindow *wwin, int req_x, int req_y, int req_width, int re
        int synth_notify = False;
        int resize;
 
+       /* if window size is guaranteed to fail - fix it to some reasonable
+        * defaults */
+       if (req_height > SHRT_MAX)
+               req_height = 480;
+
+       if (req_width > SHRT_MAX)
+               req_height = 640;
+
        resize = (req_width != wwin->client.width || req_height != wwin->client.height);
        /*
         * if the window is being moved but not resized then

